Docs to talk about intregrations (#215)
home-assistant/developers.home-assistant: docs/config_entries_config_flow_handler.md

---
title: Integration Configuration
sidebar_label: Configuration
---

> This option is currently only available for built-in components.

> Integrations can be set up via the user interface by adding support for config entries. Config entries uses the [data flow entry framework](data_entry_flow_index.md) to allow users to create entries. Components that want to support config entries will need to define a Config Flow Handler. This handler will manage the creation of entries from user input, discovery or other sources (like hassio).

Reviewer: ```suggestion
Integrations can be set up via the user interface by adding support for config entries. Config entries uses the [data flow entry framework](data_entry_flow_index.md) to allow users to create entries. Components that want to support config entries will need to define a Config Flow Handler. This handler will manage the creation of entries from user input, discovery or other sources (like Hass.io).
```

To register your config flow handler with Home Assistant, register it with the config entries `HANDLERS` registry:

```python
from homeassistant import config_entries

@config_entries.HANDLERS.register(DOMAIN)
class ExampleConfigFlow(config_entries.ConfigFlow):
```

All config flow handlers will also need to add their domain name to the `FLOWS` constant in `homeassistant/config_entries.py`.

## Discovering your config flow

> Home Assistant has a discovery integration that scans the network for available devices and services  and will trigger the config flow of the appropriate integration. Discovery is limited to uPnP and zeroconf/mDNS.

Reviewer: ```suggestion
Home Assistant has a discovery integration that scans the network for available devices and services and will trigger the config flow of the appropriate integration. Discovery is limited to UPnP and zeroconf/mDNS.
```

> To have your integration be discovered, you will have to extend the NetDisco library to be able to find your device. This is done by adding a new discoverable. [See the repository for examples of existing discoverable.](https://github.com/home-assistant/netdisco/tree/master/netdisco/discoverables)

Reviewer 2: ```suggestion
To have your integration be discovered, you will have to extend the [NetDisco library](https://github.com/home-assistant/netdisco) to be able to find your device. This is done by adding a new discoverable. [See the repository for examples of existing discoverable.](https://github.com/home-assistant/netdisco/tree/master/netdisco/discoverables)
```

Reviewer 2: I know we link it a few words later but I think it's good to have a link to the repo itself as well.

        # Handle discovery info
```

## Translations

> 

Reviewer 2: Do we need to explain Loaklise, or at least link to the dedicated page about localization? Also, do we need to tell people that even when keys are added to Lokalise they won't appear for a while until someone does a export?

Author: Done
